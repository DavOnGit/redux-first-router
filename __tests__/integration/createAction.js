import createTest from '../../__helpers__/createTest'

createTest('add routeType_COMPLETE as type to action-like object that is missing type', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch({ payload: 'foo' }),
    onComplete: function() {}
  }
})

createTest('when isAction(action) === false set to payload', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch({ foo: 'bar' }),
    onComplete: function() {}
  }
})

createTest('non object argument can be payload', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch('foo'),
    onComplete: function() {}
  }
})

createTest('null payloads allowed', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch(null),
    onComplete: function() {}
  }
})


createTest('arg as type', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch('FOO_BAR'),
    onComplete: function() {}
  }
})

createTest('arg as @@library type', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch('@@library/FOO_BAR'),
    onComplete: function() {}
  }
})

createTest('arg as type from route', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch('FIRST'),
    onComplete: function() {}
  }
})
