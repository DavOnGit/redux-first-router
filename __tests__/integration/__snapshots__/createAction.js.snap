// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - history_index 1`] = `1`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - response 1`] = `
Object {
  "payload": "foo",
  "type": "SECOND_COMPLETE",
}
`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND_COMPLETE - \\"foo\\"",
}
`;

exports[`add routeType_COMPLETE as type to action-like object that is missing type: 1 - title 1`] = `"SECOND"`;

exports[`arg as @@library type: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`arg as @@library type: 1 - history_index 1`] = `1`;

exports[`arg as @@library type: 1 - response 1`] = `
Object {
  "type": "@@library/FOO_BAR",
}
`;

exports[`arg as @@library type: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`arg as @@library type: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`arg as @@library type: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`arg as @@library type: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`arg as @@library type: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "@@library/FOO_BAR",
}
`;

exports[`arg as @@library type: 1 - title 1`] = `"SECOND"`;

exports[`arg as type from route: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`arg as type from route: 1 - history_index 1`] = `0`;

exports[`arg as type from route: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "direction": "backward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "index": 0,
    "kind": "back",
    "length": 2,
    "pathname": "/first",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/second",
            "search": "",
            "state": Object {},
            "url": "/second",
          },
          "params": Object {},
          "pathname": "/second",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "SECOND",
          "url": "/second",
        },
      ],
      "hash": "",
      "index": 1,
      "kind": "push",
      "length": 2,
      "params": Object {},
      "pathname": "/second",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "SECOND",
      "url": "/second",
    },
    "scene": "",
    "search": "",
    "status": 200,
    "url": "/first",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "FIRST",
}
`;

exports[`arg as type from route: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`arg as type from route: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`arg as type from route: 1 - routes - SECOND - onComplete 1`] = `0`;

exports[`arg as type from route: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`arg as type from route: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "backward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 0,
    "kind": "back",
    "length": 2,
    "params": Object {},
    "pathname": "/first",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/second",
            "search": "",
            "state": Object {},
            "url": "/second",
          },
          "params": Object {},
          "pathname": "/second",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "SECOND",
          "url": "/second",
        },
      ],
      "hash": "",
      "index": 1,
      "kind": "push",
      "length": 2,
      "params": Object {},
      "pathname": "/second",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "SECOND",
      "url": "/second",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "universal": false,
    "url": "/first",
  },
  "title": "FIRST",
}
`;

exports[`arg as type from route: 1 - title 1`] = `"FIRST"`;

exports[`arg as type: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`arg as type: 1 - history_index 1`] = `1`;

exports[`arg as type: 1 - response 1`] = `
Object {
  "type": "FOO_BAR",
}
`;

exports[`arg as type: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`arg as type: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`arg as type: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`arg as type: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`arg as type: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "FOO_BAR",
}
`;

exports[`arg as type: 1 - title 1`] = `"SECOND"`;

exports[`non object argument can be payload: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`non object argument can be payload: 1 - history_index 1`] = `1`;

exports[`non object argument can be payload: 1 - response 1`] = `
Object {
  "payload": "foo",
  "type": "SECOND_COMPLETE",
}
`;

exports[`non object argument can be payload: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`non object argument can be payload: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`non object argument can be payload: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`non object argument can be payload: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`non object argument can be payload: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND_COMPLETE - \\"foo\\"",
}
`;

exports[`non object argument can be payload: 1 - title 1`] = `"SECOND"`;

exports[`null payloads allowed: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`null payloads allowed: 1 - history_index 1`] = `1`;

exports[`null payloads allowed: 1 - response 1`] = `
Object {
  "payload": null,
  "type": "SECOND_COMPLETE",
}
`;

exports[`null payloads allowed: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`null payloads allowed: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`null payloads allowed: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`null payloads allowed: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`null payloads allowed: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND_COMPLETE - null",
}
`;

exports[`null payloads allowed: 1 - title 1`] = `"SECOND"`;

exports[`when isAction(action) === false set to payload: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second",
      "search": "",
      "state": Object {},
      "url": "/second",
    },
    "params": Object {},
    "pathname": "/second",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "SECOND",
    "url": "/second",
  },
]
`;

exports[`when isAction(action) === false set to payload: 1 - history_index 1`] = `1`;

exports[`when isAction(action) === false set to payload: 1 - response 1`] = `
Object {
  "payload": Object {
    "foo": "bar",
  },
  "type": "SECOND_COMPLETE",
}
`;

exports[`when isAction(action) === false set to payload: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`when isAction(action) === false set to payload: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`when isAction(action) === false set to payload: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`when isAction(action) === false set to payload: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`when isAction(action) === false set to payload: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/second",
          "search": "",
          "state": Object {},
          "url": "/second",
        },
        "params": Object {},
        "pathname": "/second",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "SECOND",
        "url": "/second",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "kind": "push",
    "length": 2,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND_COMPLETE - {\\"foo\\":\\"bar\\"}",
}
`;

exports[`when isAction(action) === false set to payload: 1 - title 1`] = `"SECOND"`;
